### PR TITLE
corelogic exports

### PR DESCRIPTION
**Pull request by nelsoneab4**
### **Link to Jira Card**

[ENG-538](https://flyreel.atlassian.net/browse/ENG-538?atlOrigin=eyJpIjoiNWFjZmE2ZDM2YzA4NDllM2FiYjJmODMxOTZmNDYzNTkiLCJwIjoiaiJ9)

### **Description**

Added cloud function to export CoreLogic inspection data. Function exports form data, photos and videos and sends a slack notification if there is an error.

### CODE REVIEW

File: src/handlers/exportInspection/exportInspection.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { full_inspection } from "../../../factories/inspection.factory";
```
Comment (johnnymac6) [inline src/handlers/exportInspection/exportInspection.unit.test.ts:1]: let's be sure to add additional unit testing with axios mocks around ensuring we are calling the corelogic API with the proper data and that we are calling the slack webhook correctly with any issues.

File: src/handlers/exportInspection/exportHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                messageAnswer
+              );
+            }
+          }
+          break;
+        }
+        case "location": {
+          fieldData.push({
+            Name: messageName,
+            Value: messageAnswer.address.replace(emojis, ""),
```
[the remark this answers is not in the document]
Comment (nelsoneab4) [reply src/handlers/exportInspection/exportHelpers.ts:81]: Added null check for address.

File: src/handlers/exportInspection/exportHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+              externalId,
+              fileName(messageAnswer),
+              messageAnswer
+            );
+          }
+          break;
+        }
+        case "video": {
+          if (allowedFileTypes.has(fileExtension(messageAnswer))) {
+            if (!photos[fileName(messageAnswer)] && message.detections) {
```
[the remark this answers is not in the document]
Comment (nelsoneab4) [reply src/handlers/exportInspection/exportHelpers.ts:54]: Removed check.

File: src/handlers/exportInspection/exportHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        .replace(spaces, "")}${
+        message.answer.iteration_id ?? "0"
+      }_${message.name.replace(punctuation, "").replace(spaces, "")}`;
+      const messageAnswer = message.answer ?? "";
+      const messageType = message.type;
+
+      switch (messageType) {
+        case "photo": {
+          if (
+            !photos[fileName(messageAnswer)] &&
```
[the remark this answers is not in the document]
Comment (nelsoneab4) [reply src/handlers/exportInspection/exportHelpers.ts:40]: Should not have any duplicate file types on our end due to id of file.

File: src/handlers/exportInspection/exportHelpers.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+const formatFormData = (
+  inspectionId: string,
+  externalId: string,
+  name: string,
+  value: string
+): any => {
+  const form = new FormData();
+  form.append("InspectionId", inspectionId);
+  form.append("UniqueId", externalId);
+  form.append(name, fs.createReadStream(value));
```
[the remark this answers is not in the document]
Comment (nelsoneab4) [reply src/handlers/exportInspection/exportHelpers.ts:113]: Changing form data formatting to happen as needed.

File: src/handlers/exportInspection/exportInspection.unit.test.ts (hunk 1 above)
Comment (nelsoneab4) [reply src/handlers/exportInspection/exportInspection.unit.test.ts:6]: Added addition unit tests to cover these.

**Review by YongDeng22 — APPROVED** [review]

File: src/handlers/exportInspection/exportHelper.unit.test.ts [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+      InspectionId: "8d59ff7264edba1ab4735b42",
+      UniqueId: "external_inspection_id",
+    });
+  });
+});
+
+describe("createFormData", () => {
+  afterEach(mock.restore);
+
+  it("should created formData", () => {
```
[the remark this answers is not in the document]
Comment (nelsoneab4) [reply src/handlers/exportInspection/exportHelper.unit.test.ts:278]: Typo fixed

**Pull request merged**